service/frontend/adminHandler.go

```diff
@@ -142,44 +145,46 @@ var (
 	}
 )
 
-// NewAdminHandler creates a thrift handler for the cadence admin service
+// NewAdminHandler creates a thrift service for the cadence admin service
 func NewAdminHandler(
-	resource resource.Resource,
+	service Service,
```

Comment:
I don't like this much, passing itself as a parameter, but I don't really want to make this a huge refactor and this isn't really significantly changing much